Add cheminformant recipe for conda-forge

HzaCode opened the pull request:
## Description
This PR adds a conda-forge recipe for `cheminformant`, a robust and high-throughput Python client for the PubChem API.

## Package Information
- **Package Name**: cheminformant
- **Version**: 2.4.2
- **License**: MIT
- **Homepage**: https://github.com/HzaCode/ChemInformant
- **PyPI**: https://pypi.org/project/cheminformant/

## Checklist
- [x] Title of this PR is meaningful: "Add cheminformant recipe"
- [x] License file is packaged (LICENSE file is present in source package)
- [x] Source is from official source (PyPI)
- [x] Package does not vendor other packages
- [x] If static libraries are linked in, the license of the static library is packaged (N/A - pure Python package)
- [x] Package does not ship static libraries (N/A - pure Python package)
- [x] Build number is 0
- [x] A tarball (url) rather than a repo is used in recipe (using PyPI tarball)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there (I am the maintainer HzaCode)
- [x] When in trouble, checked knowledge base documentation

@conda-forge/help-python ready for review!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cheminformant/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/cheminformant/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of the recipe, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).
 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18058097530](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18058097530). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/cheminformant/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/cheminformant/meta.yaml**:

 * ❌ Non noarch packages should have python requirement without any version constraints.
 * ❌ Non noarch packages should have python requirement without any version constraints.

For **recipes/cheminformant/meta.yaml**:

 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18058313662](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18058313662). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cheminformant/meta.yaml```) and found it was in an excellent condition.

HzaCode commented:
I confirm I am willing to be listed as a maintainer and will help maintain the feedstock.

HzaCode commented:
Thanks @ocefpaf!

ocefpaf commented on recipes/cheminformant/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+  sha256: 9f591b7c74feb80d8e1dba27017042c97e4c6f413205bbfc15fb1fb321accc4d
+
+build:
+  # Note: This package generates platform-specific console scripts (chemdraw.exe on Windows)
```
Please add the entry-point and make it noarch:

```yaml
  python:
    entry_points:
      - chemfetch = ChemInformant.cli:main_fetch
      - chemdraw = ChemInformant.cli:main_draw
```

eunos-1128 commented:
@HzaCode

Can you update this?

HzaCode commented:
Updated per review — bumped to 2.5.0 (pystow was dropped upstream so that concern is gone), switched to `noarch: python` with the `chemfetch`/`chemdraw` entry points, added the `pip check` test plus CLI smoke tests, and applied the `python_min` syntax. Thanks for the rebase @eunos-1128 and the review @ocefpaf!

eunos-1128 commented:
@HzaCode 

Thanks!

I converted meta.yaml to recipe.yaml with v1-format, which is encouraged in conda-forge today.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/cheminformant/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipes/cheminformant/recipe.yaml**:

 * ❌ There are 1 too many lines.  There should be one empty line at the end of the file.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/33361625136](https://github.com/conda-forge/conda-forge-webservices/actions/runs/33361625136). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cheminformant/recipe.yaml```) and found it was in an excellent condition.

eunos-1128 reviewed (approved):
LGTM
